### PR TITLE
chore(lib/tutil): move tutil and fix lints

### PR DESCRIPTION
**Pull request by corverroos**
Moves `tutil` from `e2e` to `lib`.

Also cleanup linting for e2e and other packages.

task: none

### CODE REVIEW

File: .golangci.yml [hunk 1]
```diff
@@ -82,32 +82,26 @@ linters-settings:
...
     ignoreSigs:
       - github.com/omni-network/omni/
+      - google.golang.org/grpc/status # No point wrapping gRPC/network errors.
 
 issues:
   fix: true
   exclude-rules:
-    - path: 'metrics.go'
-      linters:
-        - gochecknoglobals # Promauto metrics are global variables
```
Comment (corverroos) [inline .golangci.yml:91]: removed `gochecknoglobals` in general since we have a ton of globals in many places

File: .golangci.yml [hunk 2]
```diff
@@ -82,32 +82,26 @@ linters-settings:
...
-    - path: '(.*)(test|e2e|scripts)(.*)' # TODO: remove e2e and fix
```
Comment (corverroos) [inline .golangci.yml:92]: removed `e2e` from this list

File: .golangci.yml [hunk 3]
```diff
@@ -82,32 +82,26 @@ linters-settings:
...
+    - path: '(.*)(_test|tutil|scripts)(.*)'
       linters:         # Relax linters for both tests/scripts (non-production code)
-        - gosec        # Security not an issue here
+        - gosec        # Security not required
         - revive       # Relax revive rules
         - wrapcheck    # Wrapping not required
-        - gocognit     # Cognitive complexity not an issue here
```
Comment (corverroos) [inline .golangci.yml:97]: disabled `gocognit` since we have many big functions

File: .golangci.yml [hunk 4]
```diff
@@ -82,32 +82,26 @@ linters-settings:
...
-        - prealloc     # Prealloc not an issue here
```
Comment (corverroos) [inline .golangci.yml:98]: prealloc has lots of false positives and makes code more complex to read

**Review by Lazar955 — APPROVED** [review]

File: .golangci.yml [hunk 5]
```diff
@@ -82,32 +82,26 @@ linters-settings:
...
   fix: true
   exclude-rules:
-    - path: 'metrics.go'
-      linters:
-        - gochecknoglobals # Promauto metrics are global variables
-    - path: '(.*)(test|e2e|scripts)(.*)' # TODO: remove e2e and fix
+    - path: '(.*)(_test|tutil|scripts)(.*)'
       linters:         # Relax linters for both tests/scripts (non-production code)
-        - gosec        # Security not an issue here
+        - gosec        # Security not required
```
Comment (pavelnikolov) [inline .golangci.yml:92]: Just out of curiosity - do we fail some of the `gosec` rules?

**Pull request merged**